src/App.tsx

```diff
@@ -25,17 +40,37 @@ const initialState = {
 class App extends React.Component<{}, State> {
   public state: State = initialState
 
-  public componentWillMount = () => {
-    if (location.hash === '#sample') {
+  public componentDidMount = () => {
+    if (window.location.hash === '#sample') {
       this.setState({
         election: sampleElection,
       })
+    } else {
+      this.setState({
+        election: this.getElection(),
+      })
     }
+    Mousetrap.bind(removeElectionShortcuts, this.removeElection)
   }
 
-  public resetBallot = () => {
+  public componentWillUnount = /* istanbul ignore next */ () => {
+    Mousetrap.unbind(removeElectionShortcuts)
+  }
+
+  public getElection = () => {
+    const election = window.localStorage.getItem(electionKey)
+    return election ? JSON.parse(election) : undefined
+  }
+
+  public setElection = (election: Election) => {
+    this.setState({ election })
```

Comment:
for increased defensive coding, I'd suggest having a `reset(election)` method that both sets the `localStorage` and React component state. In other words, minimize the touchpoints between your code and `localStorage` as much as possible.

Reply:
I'm not sure what you mean. I think a diff here would more clearly explain your suggestion.

Reply:
hard for me to do a diff right now, am sneaking in a few minutes of review here and there.

Let me try to rephrase what I mean.

Right now `removeElection` and `setElection` both touch `window.localStorage` and call `setState`. Feels a bit flimsy for both to have to coordinate those two pieces of state, best if that coordination lives in only one place. One way to address this: `removeElection` just calls `setElection` with a null election. There are other ways, I don't have a strong preference, only saying that the double state-setting -- `localStorage` and React state -- should happen in only one place, with other touchpoints using a higher-level abstraction.

Reply:
Thanks for this further explanation. Very helpful.

Though, I feel like it conflicts with a previous goal you had mentioned: that `setElection` should only accept an `Election` typed object. 

I have refactored the `removeElection` method to be named `reset`.

While both `setElection` and `reset` call `setState`, they both call a different `localStorage` methods, `setItem` and `removeItem`. Is this different enough to have two "touchpoints"?